src/main/java/com/cloudbees/jenkins/support/api/PrintedContent.java

```diff
@@ -42,12 +47,29 @@ public PrintedContent(String name) {
 
     @Override
     public final void writeTo(OutputStream os) throws IOException {
-        PrintWriter out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(os, "utf-8")));
+        final PrintWriter writer = getWriter(os);
         try {
-            printTo(out);
+            printTo(writer);
         } finally {
-            out.flush();
+            writer.flush();
+        }
+    }
+
+    /**
+     * When anonymization is enabled, we create an {@link FilteredWriter} directly from
+     * the underlying {@link FilteredOutputStream} that prevents us from encoding and
+     * then immediately decoding characters written to the returned {@link PrintStream}
+     * when filtering.
+     */
+    private PrintWriter getWriter(OutputStream os) {
+        if (os instanceof WrapperOutputStream) {
+            OutputStream out = ((WrapperOutputStream) os).getRecursivelyUnderlyingStream();
+            if (out instanceof FilteredOutputStream) {
+                FilteredOutputStream filteredStream = (FilteredOutputStream) out;
```

Comment:
This is broken again. Since a new `OutputStreamSelector` is created for each `Content`, when this method is called `OutputStreamSelector#getUnderlyingStream` returns null because it has not seen any bytes. We'd want to force the text stream in this case, but I'm not sure what the cleanest approach would be.

Reply:
Yeah, I'm not exactly sure what the best approach is here for an optimization. A `ThreadLocal` could be a nice hack if needed.

Reply:
If the default behavior of the class is to use the text stream rather than the binary stream for 0 bytes, then this would be consistent and less of a hack. I'll try it out and see what it looks like.